CONTRIBUTING.md

```diff
@@ -1,13 +1,257 @@
+It is a good idea if you are unsure whether your problem or question is worthy of a Github Issue to first post it to the gitter channel.
+You may also ask questions in [Github issues](https://github.com/cocotb/cocotb/issues).
+If you don't receive any response on the gitter or a Github issue, or you want help privately, you may directly contact a [maintainer](#maintainers).
+
+
+## What to Work On
+
+There is *a lot* of work to do on this project, no matter your area of expertise or skill level.
+
+If you are a beginner there are several [Github issues](https://github.com/cocotb/cocotb/issues) marked [`good first issue`](https://github.com/cocotb/cocotb/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) you can look at.
+There are also a number of things you can work on that aren't in Github issues and don't require in-depth knowledge of the cocotb internals.
+They including the following:
+
+* Documentation improvements
+* Managing Github issues and the gitter
+* Testing and coverage improvements
+
+Cocotb is still not perfect. There are plenty of [bug fixes](https://github.com/cocotb/cocotb/issues?q=is%3Aopen+is%3Aissue+label%3Atype%3Abug) and [features])(https://github.com/cocotb/cocotb/issues?q=is%3Aopen+is%3Aissue+label%3Atype%3Afeature) that can be worked.
+Most of these are recorded as Github issues.
+
+
+### Documentation
+
+Cocotb's documentation is always open to improvements.
+Improving documentation will help users better understand and use cocotb;
+and may decrease the number of questions the gitter and Github issue page.
+Updating documentation requires the following:
+
+* A good grasp on English
+* [reStructuredText](https://docutils.sourceforge.io/rst.html)
+* [Sphinx documentation generator](https://www.sphinx-doc.org/en/master/)
+* [Markdown](https://www.markdownguide.org/)
+
+Some documentation should be located in the official documentation on [ReadTheDocs](https://docs.cocotb.org/en/latest/),
+while the rest belongs on the [Wiki](https://github.com/cocotb/cocotb/wiki).
+There are several ways to improve the documentation:
+
+* Better documenting core functionality (RTD)
+* Documenting common "gotchas" (RTD)
+* Documenting difficult and niche use cases (Wiki)
+* Documenting common design patterns (Wiki)
+* Documenting internal components (Wiki)
+
+See the documentation on [building the documentation](#Building-Documentation-Locally) and the [guidelines on submitting pull requests](#patch-requirements).
+Documentation improvements typically require no maintainer pre-approval; you can simply work on the documentation and open a pull request.
+Documentation on the Wiki does not require a pull request; any user with a Github account can contribute to it.
+Please be responsible with that power.
+
+
+### Project Management
+
+The cocotb project is [fairly popular](https://larsasplund.github.io/github-facts/verification-practices.html#frameworks) and the
+[gitter](https://gitter.im/cocotb/Lobby) and
```

Comment:
Is calling it "the gitter" common?

Reply:
Beats me. I'll update all the references to "gitter channel".

Reply:
```suggestion
[gitter channel](https://gitter.im/cocotb/Lobby) and
```